### PR TITLE
Use TLS connect for HTTPS agent

### PR DESCRIPTION
**Pull request by tom-drake**
This fixes problems I was having with the HTTPS agent. Details available under issue https://github.com/mscdex/ssh2/issues/815

### CODE REVIEW

File: lib/http-agents.js [hunk 1]
```diff
@@ -52,7 +55,7 @@ function createConnection(options, cb) {
 
 function noop() {}
 
-function decorateStream(stream) {
+function decorateHttpStream(stream) {
```
Comment (mscdex) [inline lib/http-agents.js]:
```suggestion
function decorateHttpStream(stream, options) {
```

File: lib/http-agents.js [hunk 2]
```diff
@@ -13,6 +14,7 @@ var Client;
 
     this._connectCfg = connectCfg;
     this._defaultSrcIP = (agentOptions && agentOptions.srcIP) || 'localhost';
+    this.streamDecorator = ctor === HttpAgent ? decorateHttpStream : decorateHttpsStream;
```
Comment (mscdex) [inline lib/http-agents.js]:
```suggestion
    this._streamDecorator = ctor === HttpAgent ? decorateHttpStream : decorateHttpsStream;
```

File: lib/http-agents.js [hunk 3]
```diff
@@ -42,7 +44,8 @@ function createConnection(options, cb) {
       stream.once('close', () => {
         client.end();
       });
-      cb(null, decorateStream(stream));
+
+      cb(null, this.streamDecorator(stream, options));
```
Comment (mscdex) [inline lib/http-agents.js]:
```suggestion
      cb(null, this._streamDecorator(stream, options));
```

**Comment by codenirvana** [comment]
@mscdex any update here?
I can confirm these changes did fix the issue mentioned here #815.

**Comment by acolby** [comment]
@mscdex this is an awesome project! I'm sure you're busy but this pr fixes the #815. Seems likes it's pretty low impact.

**Comment by alex2006hw** [comment]
I have a use case for this patch.  I would like to +1 for the merge.

**Comment by mscdex** [comment]
This should be resolved now with the `ssh2` rewrite. For more details see [here](https://github.com/mscdex/ssh2/issues/935).

**Pull request closed without merging**